extensions/wikia/AchievementsII/specials/templates/SpecialCustomize.tmpl.php

```diff
@@ -1,180 +1,199 @@
-		$editPlusCategoryTracks = $config->getInTrackEditPlusCategory();
+	$tracks = array();
+	foreach( $config->getInTrackStatic() as $badgeTypeId => $trackData ) {
+		if( !$config->shouldShow( $badgeTypeId ) ) {
+			continue;
+		}
+		if( $badgeTypeId != BADGE_EDIT ) {
+			$tracks[ $badgeTypeId ] = $trackData;
+		} else {
+			$tracks[ $badgeTypeId ] = $trackData;
+			$editPlusCategoryTracks = $config->getInTrackEditPlusCategory();
 
-		if( $editPlusCategoryTracks ) {
-			foreach( $editPlusCategoryTracks as $editPlusCategoryTypeId => $editPlusCategoryData ) {
-				$tracks[$editPlusCategoryTypeId] = array( 'category' => $editPlusCategoryData['category'], 'enabled' => $editPlusCategoryData['enabled'], 'laps' => $tracks[$badgeTypeId]['laps'], 'infinite' => $tracks[$badgeTypeId]['infinite'] );
+			if( $editPlusCategoryTracks ) {
+				foreach( $editPlusCategoryTracks as $editPlusCategoryTypeId => $editPlusCategoryData ) {
+					$tracks[ $editPlusCategoryTypeId ] = array( 'category' => $editPlusCategoryData[ 'category' ], 'enabled' => $editPlusCategoryData[ 'enabled' ], 'laps' => $tracks[ $badgeTypeId ][ 'laps' ], 'infinite' => $tracks[ $badgeTypeId ][ 'infinite' ] );
+				}
 			}
 		}
 	}
-}
-?>
-<?foreach($tracks as $badgeTypeId => $trackData):?>
-	<div class="customize-section" id="section<?= $badgeTypeId; ?>">
-		<div class="save-title-button">
+	?>
+	<? foreach( $tracks as $badgeTypeId => $trackData ): ?>
+		<div class="customize-section" id="section<?= $badgeTypeId; ?>">
+			<div class="save-title-button">
 			<span class="enabled-flag">
-				<?= (isset($trackData['category'])) ?
-					"<input class=\"c-enabled-flags\" type=\"checkbox\" name=\"ec_{$badgeTypeId}\" value=\"1\"". (($trackData['enabled']) ? ' checked' : null) . '/><label>' . wfMessage( 'achievements-enable-track' )->escaped() . '</label> '
-				:
-					null?>
+				<?= ( isset( $trackData[ 'category' ] ) ) ?
+					"<input class=\"c-enabled-flags\" type=\"checkbox\" name=\"ec_{$badgeTypeId}\" value=\"1\"" . ( ( $trackData[ 'enabled' ] ) ? ' checked' : null ) . '/><label>' . wfMessage( 'achievements-enable-track' )->escaped() . '</label> '
+					:
+					null ?>
 			</span>
-			<form method="POST" class="save-all">
-				<input type="hidden" name="json-data" class="c-messages" />
-				<input type="hidden" name="editToken" value="<?= Sanitizer::encodeAttribute( $editToken ); ?>" />
-				<input type="submit" value="<?= wfMessage('achievements-save')->escaped() ?>" onclick="Achievements.AchPrepareData(false, '<?= $badgeTypeId; ?>');"/>
-			</form>
-		</div>
-		<h2><?= wfMessage( $config->getTrackNameKey($badgeTypeId), (isset($trackData['category']) ? $trackData['category'] : null) )->escaped(); ?></h2>
+				<form method="POST" class="save-all">
+					<input type="hidden" name="json-data" class="c-messages"/>
+					<input type="hidden" name="editToken" value="<?= Sanitizer::encodeAttribute( $editToken ); ?>"/>
+					<input type="submit" value="<?= wfMessage( 'achievements-save' )->escaped() ?>"
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "achievements-save" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)